pulsar/producer_partition.go

```diff
@@ -298,7 +306,8 @@ func (p *partitionProducer) reconnectToBroker() {
 		d := backoff.Next()
 		p.log.Info("Reconnecting to broker in ", d)
 		time.Sleep(d)
-
+		atomic.AddUint64(&p.epoch, 1)
+		p.log.WithField("epoch", atomic.LoadUint64(&p.epoch))
```

Comment:
The epoch field is going to log the same value every time (the first value). Can epoch just be logged after the producer is created? Updating the field value requires creating a new logger and doing this would require locking to access the log which is probably not what we want.

Reply:
I see, just removed the epoch from the logger and add a one-line debug log to print the current epoch value.